### ISSUE TITLE
Duplicate metrics generated although replication factor is 1

### ISSUE THREAD

**Issue by mwtzzz** [grobian/carbon-c-relay#295]
I need some help figuring out why duplicate metrics are being generated although my hashing scheme is set to a replication factor of 1. I have a backend cluster with twelve hosts and some metrics with the same metric path are getting updated on two of them. Here is an example of such a metric:
`/media/ephemeral0/carbon/storage/whisper/central/dao/save/BudgetWrite/sum_all/hosts.wsp`

My relay layer conf looks like this:
```
cluster debug
  forward
    127.0.0.1:62000 proto udp
  ;

cluster radar-be
  fnv1a_ch 
    radar-be-a:1905=a 
    radar-be-b:1905=b 
    radar-be-c:1905=c 
    <...>
  ;
match ^agg\. send to radar-be;
match ^(us-east-1\.)?carbon\. send to radar-be;
match ^agg\. send to blackhole stop;
match ^(us-east-1\.)?carbon\. send to blackhole stop;

aggregate
    ^(.+)\.sum\.([^.]+)$
  every 60 seconds
  expire after 90 seconds
  timestamp at end of bucket
  compute sum write to agg.\1.sum_all.hosts
  ;
aggregate
    ^(us-east-1\.)?geocoder\.request\.(XC|UNKNOWN|AVAILS|NEPTUNE_SUPERMEDIA)\.[^.]+\.[^.]+\.sum\.
[^.]+$
  every 60 seconds
  expire after 90 seconds
  timestamp at end of bucket
  compute sum write to agg.geocoder.appid.\2
  ;

match ^(us-east-1\.)?atlantic_exchange\.adgroup\. send to blackhole stop;
match ^(us-east-1\.)?nise\.(campaigns|adgroups)\. send to blackhole stop;

match * send to radar-be;
```
My backend relay (port 1905) looks like this:
```
cluster debug
  forward
    127.0.0.1:62001 proto udp
  ;

cluster cache
  fnv1a_ch
    127.0.0.1:2105=2105
    127.0.0.1:2107=2107
    127.0.0.1:2109=2109
    127.0.0.1:2111=2111
    127.0.0.1:2113=2113
    127.0.0.1:2115=2115
  ;

match * send to debug;
aggregate
    ^(us-east-1\.)?agg\.(.+)$
  every 60 seconds
  expire after 90 seconds
  timestamp at start of bucket
  compute sum write to \2
  ;

match ^agg\. send to blackhole stop;
match * send to cache stop;
```

Any thoughts?

**Comment by grobian** [comment]
what version of the relay are your running?

**Comment by grobian** [comment]
I think I found the culprit, could you try and see if current head still exhibits this behaviour?

**Comment by mwtzzz** [comment]
I'm running `carbon-c-relay v3.1` + the patch you had made a couple months back to fix the aggregator loop issue..

**Comment by grobian** [comment]
I see.  Are these metrics filled with all points on both machines?  Or are the datapoints scattered between the machines?

**Comment by mwtzzz** [comment]
Datapoints appear to be scattered between the machines. It's difficult to tell for sure how much duplication is occuring, because each machine has about 1.5million wsp files.

**Comment by grobian** [comment]
I assume you redacted your config, is the metric you pasted the path for one that would be generated by the aggregator rule producing agg.\1.sum_all.hosts?

**Comment by mwtzzz** [comment]
Yes, I redacted it, the only thing obscured are hostnames. I'm pretty sure this is NOT the fault of carbon-c-relay. I restarted services, made sure the configs were consistent, and the handful of metrics I was watching are no longer duplicated. It's possible there were some extra relay processes or something like that happening. 

I'm going to say let's close this issue for now. If we can get buckytools to support fnv1a_ch hash, that would make it easier to monitor for duplicates.
